azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java

```diff
@@ -307,34 +309,67 @@ public void testAssignExecutorInvalidExecution() throws Exception {
         .hasMessageContaining("non-existent execution");
   }
 
-
   @Test
   public void testFetchActiveFlowsExecutorAssigned() throws Exception {
 
-    // Upload flow1, executor assigned
-    final ExecutableFlow flow1 = TestUtils.createTestExecutableFlow("exectest1", "exec1");
-    this.executionFlowDao.uploadExecutableFlow(flow1);
     final Executor executor = this.executorDao.addExecutor("test", 1);
-    this.assignExecutor.assignExecutor(executor.getId(), flow1.getExecutionId());
 
-    // Upload flow2, executor not assigned
-    final ExecutableFlow flow2 = TestUtils.createTestExecutableFlow("exectest1", "exec2");
-    this.executionFlowDao.uploadExecutableFlow(flow2);
+    final ExecutableFlow flow1 = createExecutionAndAssign(Status.PREPARING, executor);
+    // flow2 is not assigned
+    final ExecutableFlow flow2 = createExecution(Status.PREPARING);
+    final ExecutableFlow flow3 = createExecutionAndAssign(Status.RUNNING, executor);
+    final ExecutableFlow flow4 = createExecutionAndAssign(Status.SUCCEEDED, executor);
 
-    final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> activeFlows1 =
+    final Executor executor2 = this.executorDao.addExecutor("test2", 2);
+    // flow5 is assigned to an executor that is then removed
+    final ExecutableFlow flow5 = createExecutionAndAssign(Status.RUNNING, executor2);
+    this.executorDao.removeExecutor(executor2.getHost(), executor2.getPort());
+
+    final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> activeFlows =
         this.fetchActiveFlowDao.fetchActiveFlows();
 
-    assertThat(activeFlows1.containsKey(flow1.getExecutionId())).isTrue();
-    assertThat(activeFlows1.get(flow1.getExecutionId()).getFirst().getExecutor().isPresent())
-        .isTrue();
-    assertThat(activeFlows1.containsKey(flow2.getExecutionId())).isTrue();
-    assertThat(activeFlows1.get(flow2.getExecutionId()).getFirst().getExecutor().isPresent())
-        .isFalse();
+    assertFound(activeFlows, flow1, true);
+    assertNotFound(activeFlows, flow2, "Returned a queued execution");
```

Comment:
This assert reproduces the bug. It fails if run without the fix in FetchActiveFlowDao.